src/render/camera.cpp

```diff
@@ -1,31 +1,54 @@
 #include "camera.h"
 
-cr::camera::camera(const glm::vec3 position, const glm::vec3 look_at, float fov)
-    : look_at(look_at), fov(fov)
+cr::camera::camera(glm::vec3 position, float fov) :
+position(position), fov(fov), _cached_matrix(1)
 {
-    const auto theta       = fov * 3.141592f / 180.0f;
-    const auto half_height = tanf(theta / 2.f);
-    const auto half_width  = half_height;
-
-    constexpr auto up = glm::vec3(0, 1, 0);
-    const auto     w  = glm::normalize(look_at - position);
-    const auto     u  = glm::normalize(glm::cross(up, w));
-    const auto     v  = glm::cross(w, u);
-
-    this->horizontal = u * half_width;
-    this->vertical   = v * half_height;
-    this->position   = position;
-    this->center     = position + w;
+
 }
-cr::ray cr::camera::get_ray(float x, float y, const cr::camera &camera)
+
+glm::mat4 cr::camera::mat4() const noexcept
+{
+    return _cached_matrix;
+}
+
+cr::ray cr::camera::get_ray(float x, float y)
```

Comment:
Do we want another function for ortho projection, and other types of camera projects?